examples/master.js

```diff
@@ -315,12 +324,49 @@ function onPeerConnectionFailed(remoteClientId, printLostConnectionLog = true, h
     }
 }
 
+/**
+ * Fetches ICE servers, caching them to prevent redundant API calls.
+ * If the cached TURN servers are still valid, it returns them instead of making a new request.
+ * @param {Object} formValues - Configuration settings from the UI.
+ * @returns {Array} List of ICE servers.
+ */
+async function getIceServersWithCaching(formValues) {
+    const role = ROLE;
+
+    // Check if cached TURN servers are still valid
+    if (Date.now() < master.turnServerExpiry) {
+        return master.iceServers;
+    }
+    console.log(`[${role}]`, 'Fetch new ICE servers');
+
+    /** @type {RTCIceServer[]} */
+    const iceServers = [];
+
+    // Add the STUN server unless it is disabled
+    if (!formValues.natTraversalDisabled && !formValues.forceTURN && (formValues.sendSrflxCandidates || formValues.sendPrflxCandidates)) {
+        iceServers.push({ urls: `stun:stun.kinesisvideo.${formValues.region}.amazonaws.com:443` });
+    }
+
+    // Add the TURN servers unless it is disabled
+    if (!formValues.natTraversalDisabled && !formValues.forceSTUN && formValues.sendRelayCandidates) {
+        const [turnServers, turnServerExpiryMillis] = await master.channelHelper.fetchTurnServers();
+        master.turnServerExpiry = turnServerExpiryMillis - iceServerRefreshGracePeriodSec * 1000;
+        iceServers.push(...turnServers);
+    }
+    console.log(`[${role}]`, 'ICE servers:', iceServers);
+
+    master.iceServers = iceServers;
+    return iceServers;
+}
+
 function stopMaster() {
     const role = ROLE;
     try {
         console.log(`[${role}] Stopping ${role} connection`);
         master.sdpOfferReceived = true;
 
+        // Prevent it from reopening when trying to close
```

Comment:
// Wait until the close is complete

Reply:
This removes the callback that reopens the connection.

If we don't do that, when close is called, the close callback will trigger which will reopen it.